curator-recipes/src/main/java/org/apache/curator/framework/recipes/cache/PathChildrenCache.java

```diff
@@ -448,7 +448,10 @@ public boolean clearDataBytes(String fullPath, int ifVersion)
         {
             if ( (ifVersion < 0) || (ifVersion == data.getStat().getVersion()) )
             {
-                data.clearData();
+                if ( data.getData() != null )
+                {
+                    currentData.replace(fullPath, data, new ChildData(data.getPath(), data.getStat(), null));
+                }
```

Comment:
@Randgalt this is the replacement for `clearData()`.  Instead of the ChildData object being mutable, we just replace the ChildData entirely with a copy that has no data.